core/server/master/src/main/java/alluxio/master/AlluxioMasterProcess.java

```diff
@@ -165,7 +239,13 @@
       MasterUtils.createMasters(mJournalSystem, mRegistry, mSafeModeManager);
 
       // Create config checker
+      mMasterConfigChecker = new ServerConfigurationChecker();
       mWorkerConfigChecker = new ServerConfigurationChecker();
+      mClock = new SystemClock();
+      mExecutorServiceFactory = ExecutorServiceFactories
+          .fixedThreadPoolExecutorServiceFactory(Constants.META_MASTER_NAME, 2);
+      mMetaMasterClient = new MetaMasterMasterClient(MasterClientConfig.defaults());
+      setMasterIdAndHostname();
```

Comment:
This looks like this is called whether or not this master is leader or standby. Does this mean the leader must connect to itself to get a master id? Will this work?

Reply:
I guess the failed test may be because of this issue. I will change this part. When a master is a standby master, it will set its id.